### ISSUE TITLE
`DeferredResource._ebChild` uselessly propagates errors

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8192]
|[<img alt="mithrandi's avatar" src="https://avatars.githubusercontent.com/u/856430?s=50" width="50" height="50">](https://github.com/mithrandi)| @mithrandi reported|
|-|-|
|Trac ID|trac#8192|
|Type|defect|
|Created|2016-02-02 08:54:26Z|
|Branch|https://github.com/twisted/twisted/tree/suppress-error-8192|

The failure is returned, but since nothing has a reference to the deferred constructed in `render`, no future errback can ever be added to handle it. The failure just shouldn't be returned.

<details><summary>Searchable metadata</summary>

```
trac-id__8192 8192
type__defect defect
reporter__mithrandi mithrandi
priority__normal normal
milestone__None None
branch__branches_suppress_error_8192 branches/suppress-error-8192
branch_author__mithrandi mithrandi
status__closed closed
resolution__fixed fixed
component__web web
keywords__None None
time__1454403266803075 1454403266803075
changetime__1455940183083602 1455940183083602
version__None None
owner__mithrandi mithrandi

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="mithrandi's avatar" src="https://avatars.githubusercontent.com/u/856430?s=50" width="50" height="50">](https://github.com/mithrandi)<a name="note_1"></a>|@mithrandi commented|
|-|-|

(In [46830]) Branching to suppress-error-8192.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban set owner to @mithrandi|
|-|-|

Changes look good. Thanks!

Just minor comments.

1. It needs a news file fragment.

2. Why the change to SynchronousTestCase ?

3. For the test docstring, maybe have something like


```
If the L{Deferred} fails, L{DeferredResource} does not cause an
unhandled error to be logged as the error is reported using
C{processingFailed} on the request.
```

and the test extended to check that processing failed is called on the resource.

I don't see any test for that... but is 1AM here :(


Please address my comments and merge.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="mithrandi's avatar" src="https://avatars.githubusercontent.com/u/856430?s=50" width="50" height="50">](https://github.com/mithrandi)<a name="note_4"></a>|@mithrandi commented|
|-|-|

Thanks for the review :)

 1. Added.

 2. I think it's better to use `SynchronousTestCase` for all test cases that do not require the extra logic in `TestCase` (ie. asynchronous test handling); such as this one.

 3. Implemented this change (there was, indeed, no coverage for this case).

Spinning the builders one last time before merging.

**Comment by twisted-trac** [comment]
|[<img alt="mithrandi's avatar" src="https://avatars.githubusercontent.com/u/856430?s=50" width="50" height="50">](https://github.com/mithrandi)<a name="note_5"></a>|@mithrandi set status to `closed`|
|-|-|

(In [46836]) Merge suppress-error-8192: Suppress an unhandled error in DeferredResource.

Author: mithrandi
Reviewer: adiroiban
Fixes: [#8192](https://github.com/twisted/twisted/issues/8192)
